src/auth.c

```diff
@@ -51,17 +51,32 @@
 
 #include <errno.h>
 
-#if defined(HAVE_POLKIT) && defined(SO_PEERCRED)
+#if defined(HAVE_POLKIT) && (defined(SO_PEERCRED) || defined(LOCAL_PEERCRED))
 
 #include <polkit/polkit.h>
 #include <stdbool.h>
 
+#ifdef __FreeBSD__
```

Comment:
I don't know much about socket authentication, but what about macOS? Is the `struct xucred` only used on FreeBSD? A quick Google search implies that it's also used on Mac. Would you mind explaining what `ucred` vs `xucred` is about?

Reply:
`xucred` is the ABI-stable userland export version (subset) of `ucred` -- if you have the former then the latter is considered kernel-only and it will populate an `xucred` for things in userland that want it.

Reply:
macOS also seems to have LOCAL_PEERCRED,  but I don't know what the polkit-on-macOS situation is like so I haven't bothered trying to expand it in that direction.  Likely it also wants the xucred version of this.